lib/chat_api/reporting.ex

```diff
@@ -6,6 +6,7 @@ defmodule ChatApi.Reporting do
   import Ecto.Query, warn: false
   alias ChatApi.{Repo, Conversations.Conversation, Messages.Message, Users.User}
 
+  @spec count_messages_by_date(binary(), map()) :: [Message.t()]
```

Comment:
a bunch of these `count_x_by_date` methods actually return in the format:
```elixir
[%{date: '2020-09-01', count: 5}, %{date: '2020-09-02', count: 8}, ...]
```
rather than e.g. `Message.t()`

feel free to just use `any` if this module is too much work though!

Reply:
will fix it.